### DISCUSSION TITLE
dlib installation

### DISCUSSION THREAD

**Discussion by sreeharshabalaji** [datasciencecompleteguide/Machine-Learning#12] [Q&A] [answered]
getting error
Running setup.py clean for dlib
Failed to build dlib
Installing collected packages: dlib
    Running setup.py install for dlib ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\balaj\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\balaj\\AppData\\Local\\Temp\\pip-install-dk67x_x1\\dlib_79503447deb241a7b1a4d111bb64c31e\\setup.py'"'"'; __file__='"'"'C:\\Users\\balaj\\AppData\\Local\\Temp\\pip-install-dk67x_x1\\dlib_79503447deb241a7b1a4d111bb64c31e\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\balaj\AppData\Local\Temp\pip-record-hue_jc_7\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\balaj\appdata\local\programs\python\python39\Include\dlib'

**Comment by datasciencecompleteguide** [comment]
> 
> 
> getting error
> Running setup.py clean for dlib
> Failed to build dlib
> Installing collected packages: dlib
> Running setup.py install for dlib ... error
> ERROR: Command errored out with exit status 1:
> command: 'c:\users\balaj\appdata\local\programs\python\python39\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\Users\balaj\AppData\Local\Temp\pip-install-dk67x_x1\dlib_79503447deb241a7b1a4d111bb64c31e\setup.py'"'"'; **file**='"'"'C:\Users\balaj\AppData\Local\Temp\pip-install-dk67x_x1\dlib_79503447deb241a7b1a4d111bb64c31e\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(**file**) if os.path.exists(**file**) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, **file**, '"'"'exec'"'"'))' install --record 'C:\Users\balaj\AppData\Local\Temp\pip-record-hue_jc_7\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\balaj\appdata\local\programs\python\python39\Include\dlib'

Have you tried this: 
https://www.youtube.com/watch?v=eaEndTeUiSU

**Comment by sreeharshabalaji** [comment]
Yes

**Comment by sreeharshabalaji** [comment]
Could you please help us
Rajeev

**Comment by sreeharshabalaji** [comment]
Because we project demonstration on day after tomorrow so please help me

**Comment by datasciencecompleteguide** [comment, marked as answer]
This error shows the installation is not done properly.
So, I would recommend you to install it again.
First undo all the previous changes and installation done and try it in fresh way.

**Comment by sreeharshabalaji** [comment]
Should I install visual studio again

**Reply by datasciencecompleteguide** [reply]
yes, please follow the complete video step by step.
I know its too much long process but even a single missing step can stop the complete process.
first, undo all the changes by uninstalling the material u have installed and then try it a fresh way.

**Comment by sreeharshabalaji** [comment]
Again I am facing same issue

**Reply by datasciencecompleteguide** [reply]
****use the lower version of python like 3.6.9
because the wheel files are yet not available of python v9
and c++ dev. tools are necessary. Make sure u have installed them while installing visual studio

**Comment by sreeharshabalaji** [comment]
Ok, I will check.
